### DISCUSSION TITLE
Run custom rule without PSRule modules

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/PSRule#1553] [Q&A]
Hi, I am trying to run the custom rule for the timebeing without using PSRule modules like PSRule.Rules.Azure, PSRule.Rules.CAF etc.  Is it possible to do that through ps-rule-asset@2 task?  When I include default modules, I could see that my custom rule is getting triggered. But when I remove those modules, the custom rule is not getting triggered.  I am using the below binding configuration to use Type parameter. 
# Configure binding options
binding:
  targetType:
  - 'resourceType'
  - 'type'

**Comment by BernieWhite** [comment]
@jegadeesan-ponnusamy Thanks for the question. Yes you can run PSRule without additional modules such as `PSRule.Rules.Azure`. However specifically, if you are writing rules for testing Azure Bicep or ARM templates you will need to reference `PSRule.Rules.Azure`.

This is because support for ARM template and Bicep files is included in this module.

**Reply by ghost** [reply]
@BernieWhite - Thanks for your input. I am able to get the expected behaviour by using "Rule.Include".

**Reply by BernieWhite** [reply]
@jegadeesan-ponnusamy yes. If you want to use PSRule.Rules.Azure for reading your IaC files but only run your rules.

You would update Rule.Include or Rule.Tag to specify your rules.

**Reply by BernieWhite** [reply]
@jegadeesan-ponnusamy Did this help answer the question?
